components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/handler/SAMLAssertionFinderImpl.java

```diff
@@ -76,42 +79,36 @@ public Assertion findByID(String id) throws IdentitySAML2QueryException {
      * @throws  IdentitySAML2QueryException If unable to collect assertions from database
      */
     public List<Assertion> findBySubject(String subject) throws IdentitySAML2QueryException {
-        List<Assertion> assertions = new ArrayList<Assertion>();
+
+        List<Assertion> assertions;
         Connection connection = null;
         PreparedStatement preparedStatement = null;
         ResultSet resultSet = null;
         try {
             connection = JDBCPersistenceManager.getInstance().getDBConnection();
-            String selectBySubjectQuery = "SELECT SAML2_ASSERTION FROM IDN_SAML2_ASSERTION_STORE WHERE SAML2_SUBJECT =?";
-            preparedStatement = connection.prepareStatement(selectBySubjectQuery);
+            preparedStatement = connection.prepareStatement(DBUtil.SELECT_BY_SUBJECT_QUERY);
             preparedStatement.setString(1, subject);
             resultSet = preparedStatement.executeQuery();
-            int index = 1;
-            while (resultSet.next()) {
-                String assertionString = resultSet.getString(index);
-                assertions.add((Assertion) SAMLQueryRequestUtil.unmarshall(assertionString));
-                log.debug("Assertions retrieved from database for the subject: " + subject);
+            assertions = extractAssertions(resultSet);
+            if (assertions.size() > 0) {
```

Comment:
If we only have a debug level log in this block - we need to check whether debug is enabled - before this if condition.